payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java

```diff
@@ -459,7 +460,37 @@ private PriorPaymentControlResult getPluginCreditResult(final PaymentControlCont
         if (prop == null) {
             return ImmutableMap.<UUID, BigDecimal>of();
         }
-        return (Map<UUID, BigDecimal>) prop.getValue();
+        // The deserialization may not recreate the map we expect, i.e Map<UUID, BigDecimal>, so we convert each key/value by hand
+        // See https://github.com/killbill/killbill/issues/1453
+        final Map<UUID, BigDecimal> res = new HashMap<>();
+        final Map m = (Map) prop.getValue();
+        for (final Object k : m.keySet()) {
+            UUID uuid;
+            if (k instanceof String) {
+                uuid = UUID.fromString((String) k);
```

Comment:
You could maybe be a bit more generic:

```
try {
  uuid = UUID.fromString(v.toString());
} catch () { // throw }
```

That way, we'd handle other types as well (e.g. `CharSequence`).

Reply:
The only use case we really have is one when we ned to convert String -> UUID - i'd rather keep the code as it is.